sw/device/silicon_creator/rom_ext/sigverify_keys.c

```diff
@@ -0,0 +1,186 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/rom_ext/sigverify_keys.h"
+
+#include <assert.h>
+#include <stddef.h>
+
+#include "sw/device/lib/base/bitfield.h"
+#include "sw/device/lib/base/hardened.h"
+#include "sw/device/silicon_creator/lib/drivers/rnd.h"
+#include "sw/device/silicon_creator/lib/sigverify/sigverify.h"
+
+/**
+ * Determines whether a key is valid in the RMA life cycle state.
+ *
+ * Only test and production keys that are valid in the RMA life cycle state.
+ *
+ * @param key_type Type of the key.
+ * @param key_index Index of the key.
+ * @return The result of the operation.
+ */
+static rom_error_t key_is_valid_in_lc_state_rma(sigverify_key_type_t key_type,
+                                                size_t key_index) {
+  switch (launder32(key_type)) {
+    case kSigverifyKeyTypeTest:
+      HARDENED_CHECK_EQ(key_type, kSigverifyKeyTypeTest);
+      return kErrorOk;
+    case kSigverifyKeyTypeProd:
+      HARDENED_CHECK_EQ(key_type, kSigverifyKeyTypeProd);
+      return kErrorOk;
+    case kSigverifyKeyTypeDev:
+      HARDENED_CHECK_EQ(key_type, kSigverifyKeyTypeDev);
+      return kErrorSigverifyBadKey;
```

Comment:
I think this is the same condition as in the ROM: I'm not sure why we are disqualifying the DEV key in the RMA state.

Reply:
Created an issue to review this https://github.com/lowRISC/opentitan/issues/17364

Reply:
because dev keys are only for the dev life cycle state.